src/commands/filter_parser.cc

```diff
@@ -802,28 +833,38 @@ absl::StatusOr<std::unique_ptr<query::Predicate>> FilterParser::ParseTextTokens(
 
 // Parsing rules:
 // 1. Predicate evaluation is done with left-associative grouping while the OR
-// operator has higher precedence than the AND operator. precedence. For
+// operator has lower precedence than the AND operator. precedence. For
 // example: a & b | c & d is evaluated as (a & b) | (c & d).
-// 2. Field name is always preceded by '@' and followed by ':'.
-// 3. A numeric field has the following pattern: @field_name:[Start,End]. Both
+// 2. Brackets have the higest Precedence of all the operators -> () > AND > OR.
+// example a & ( b | c ) & d is evaluated as AND (a, OR(b , c), d)
+// 3. If a bracket has atleast 2 terms it will be evaluated as a
+// separate nested structure.
+// 4. If a bracket has no terms it will be evaluted to false.
```

Comment:
I don't see this implemented. Can we add this check when we are handling the result of a parsed parenthesis?